src/main/java/seedu/address/model/filter/OwnerNameContainsFilterWordPredicate.java

```diff
@@ -0,0 +1,24 @@
+package seedu.address.model.filter;
+
+import seedu.address.commons.util.StringUtil;
+import seedu.address.model.pet.Pet;
+
+public class OwnerNameContainsFilterWordPredicate extends FilterByContainsFilterWordPredicate {
+
+    public OwnerNameContainsFilterWordPredicate(String filterWord) {
+        super(filterWord);
+    }
+
+    @Override
+    public boolean test(Pet pet) {
+        return StringUtil.nameContainsWordsIgnoreCase(pet.getOwnerName().toString(), getFilterWord());
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof OwnerNameContainsFilterWordPredicate // instanceof handles nulls
+                && getFilterWord().equals(((OwnerNameContainsFilterWordPredicate) other)
+                .getFilterWord())); // state check
```

Comment:
Maybe splitting this up would be better, since it is quite hard to read! I noticed the same issue in your other `equals` methods as well.

Reply:
This was how the `NameContainsKeywordPredicate` class in AB3 implemented its equals method. However, I get your point and I will try to make it more readable!